scanner/cmd/scannerctl/main.go

```diff
@@ -38,107 +38,74 @@ func main() {
 		utils.CrashOnError(os.Setenv(mtls.CertFilePathEnvName, filepath.Join(*certsPath, mtls.ServiceCertFileName)))
 		utils.CrashOnError(os.Setenv(mtls.KeyFileEnvName, filepath.Join(*certsPath, mtls.ServiceKeyFileName)))
 	}
+
 	// Extract basic auth username and password.
-	var username, password string
+	auth := authn.Anonymous
+	if *basicAuth == "" {
+		*basicAuth = os.Getenv(authEnvName)
+	}
 	if *basicAuth != "" {
 		var ok bool
-		username, password, ok = strings.Cut(*basicAuth, ":")
+		u, p, ok := strings.Cut(*basicAuth, ":")
 		if !ok {
 			log.Fatalf("Invalid auth: %q", *basicAuth)
```

Comment:
using the safer env var method improperly can still expose creds in the logs here. Can we update this to just specify the expected format `username:password`?

Reply:
0a184683